shipment_advice_bill_auto_complete/models/account_move.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    shipment_advice_auto_complete_id = fields.Many2one("shipment.advice", store=False)
+
+    @api.onchange("shipment_advice_auto_complete_id")
+    def _onchange_shipment_auto_complete(self):
+        shipment = self.shipment_advice_auto_complete_id
+        self.shipment_advice_auto_complete_id = False
+        if not shipment.shipment_type == "incoming":
+            return
```

Comment:
use `!=` instead of `not ... ==`
```suggestion
        if self.move_type not in ("in_invoice", "out_refund") or shipment.shipment_type != "incoming":
            return
```